src/Instrumentation/Psr15/src/Psr15Instrumentation.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Psr15;
+
+use OpenTelemetry\API\Common\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Common\Instrumentation\Globals;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Throwable;
+
+/**
+ * @psalm-suppress ArgumentTypeCoercion
+ */
+class Psr15Instrumentation
+{
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation('io.opentelemetry.contrib.php.psr15');
+
+        /**
+         * Create a span for each psr-15 middleware that is executed.
+         */
+        hook(
+            MiddlewareInterface::class,
+            'process',
+            static function (MiddlewareInterface $middleware, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
+                $span = $instrumentation->tracer()->spanBuilder(sprintf('%s::%s', $class, $function))
+                    ->setAttribute('code.function', $function)
+                    ->setAttribute('code.namespace', $class)
+                    ->setAttribute('code.filepath', $filename)
+                    ->setAttribute('code.lineno', $lineno)
+                    ->startSpan();
+
+                Context::storage()->attach($span->storeInContext(Context::getCurrent()));
+            },
+            static function (MiddlewareInterface $middleware, array $params, ?ResponseInterface $response, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
```

Comment:
Instead of using a nullish coalescer here, you could move it after the `if (!$scope) {...}` block entirely, since if it is null it won't have any `detach` method to call and you're going to return it anyway.